lib/stripe/stripe_client.rb

```diff
@@ -456,19 +459,22 @@ def self.maybe_gc_connection_managers
 
       query_params, path = merge_query_params(query_params, path)
 
-      headers = request_headers(api_key, method)
+      headers = request_headers(api_key, method, api_mode)
                 .update(Util.normalize_headers(headers))
+
+      headers.delete("Content-Type") if api_mode == :preview && body_params.nil?
```

Comment:
Instead of calling delete here can we add a condition to `request_headers` and not add the header at all?